### PR TITLE
add docs for ws transport

### PR DESCRIPTION
**Pull request by karen-stepanyan**
[PDI-33](https://smartcontract-it.atlassian.net/browse/PDI-33)

[PDI-33]: https://smartcontract-it.atlassian.net/browse/PDI-33?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/v3-ea-components/transport-types/websocket-transport.md [hunk 1]
```diff
@@ -1,3 +1,227 @@
...
+      url: (context) => context.adapterConfig.WS_API_ENDPOINT,
+      options: async (context) => ({
+        headers: {
+          'x-auth-token': context.adapterConfig.API_KEY
+        },
+      })
+```
+
+### Handling opened connection (optional)
+**open** method of **handlers** object is executed once the websocket connection is successfully opened. In some scenarios it is necessary to perform logic when connecting to websocket but before sending subscription messages. It accepts raw websocket connection and adapter context as arguments and should return a Promise that either resolves or rejects with error message
```
Comment (boxhock) [inline docs/v3-ea-components/transport-types/websocket-transport.md:62]:
```suggestion
The **open** method of the **handlers** object is executed once the websocket connection is successfully opened. In some scenarios it is necessary to perform logic when connecting to websocket, but before sending subscription messages. It accepts a raw websocket connection and adapter context as arguments and should return a Promise that either resolves or rejects with error message
```

> return a Promise that either resolves

Resolves with what?

Comment (karen-stepanyan) [reply docs/v3-ea-components/transport-types/websocket-transport.md:62]: with anything or nothing. The important thing is that it resolves.

File: docs/v3-ea-components/transport-types/websocket-transport.md [hunk 2]
```diff
@@ -1,3 +1,227 @@
...
+      },
+      builders: {
+        subscribeMessage: (params) => ({
+          action: 'subscribe',
+          symbols: [`${params.base}.${params.quote}`],
+        }),
+        unsubscribeMessage: (params) => ({
+          action: 'unsubscribe',
+          symbols: [`${params.base}.${params.quote}`],
+        }),
```
[the remark this answers is not in the document]
Comment (karen-stepanyan) [reply docs/v3-ea-components/transport-types/websocket-transport.md:38]: there is a section where those are described. isn't that enough ?

File: docs/v3-ea-components/transport-types/websocket-transport.md (hunk 1 above)
Comment (boxhock) [reply docs/v3-ea-components/transport-types/websocket-transport.md]: Got it, I think it's fine then

**Review by boxhock — DISMISSED** [review]

**Review by alejoberardino — APPROVED** [review]

**Pull request merged**